### ISSUE TITLE
File permissions in linux tar are incorrect, should not have owner executable except for pwsh

### ISSUE THREAD

**Issue by rhubarb-geek-nz** [PowerShell/PowerShell#18899]
### Prerequisites

- [X] Write a descriptive title.
- [X] Make sure you are able to repro it on the [latest released version](https://github.com/PowerShell/PowerShell/releases)
- [X] Search the existing issues.
- [X] Refer to the [FAQ](https://github.com/PowerShell/PowerShell/blob/master/docs/FAQ.md).
- [X] Refer to [Differences between Windows PowerShell 5.1 and PowerShell](https://docs.microsoft.com/powershell/scripting/whats-new/differences-from-windows-powershell).

### Steps to reproduce

File permissions in `linux tar` are incorrect, should not have owner executable except for `pwsh`

The `pwsh` permission for `pwsh` is wrong for `7.3.0` but this is a known issue that is fixed in `7.3.1` but the remaining files are correct in `7.3.0`

In 7.3.1 `pwsh` is correct but most of the remaining files are wrong

### Expected behavior

```console
$ curl --silent --location https://github.com/PowerShell/PowerShell/releases/download/v7.3.0/powershell-7.3.0-linux-arm64.tar.gz | tar tvfz - | grep System.dll
-rw-r--r-- cloudtest/cloudtest   16016 2022-11-07 22:24 ./ref/System.IO.FileSystem.dll
-rw-r--r-- cloudtest/cloudtest   49784 2022-11-07 22:24 ./ref/System.dll
-rw-r--r-- cloudtest/cloudtest   15992 2022-11-07 22:24 ./ref/System.IO.Compression.FileSystem.dll
-rw-r--r-- cloudtest/cloudtest   16016 2022-11-07 22:23 ./System.IO.FileSystem.dll
-rw-r--r-- cloudtest/cloudtest   50816 2022-11-07 22:23 ./System.dll
-rw-r--r-- cloudtest/cloudtest    16016 2022-11-07 22:23 ./System.IO.Compression.FileSystem.dll
```

### Actual behavior

```console
$ curl --silent --location https://github.com/PowerShell/PowerShell/releases/download/v7.3.1/powershell-7.3.1-linux-arm64.tar.gz | tar tvfz - | grep System.dll
-rwxr--r-- cloudtest/cloudtest   15992 2022-11-23 03:27 ref/System.IO.Compression.FileSystem.dll
-rwxr--r-- cloudtest/cloudtest   49784 2022-11-23 03:27 ref/System.dll
-rwxr--r-- cloudtest/cloudtest   16016 2022-11-23 03:27 ref/System.IO.FileSystem.dll
-rwxr--r-- cloudtest/cloudtest    50816 2022-10-18 16:10 System.dll
-rwxr--r-- cloudtest/cloudtest    16016 2022-10-18 16:10 System.IO.Compression.FileSystem.dll
-rwxr--r-- cloudtest/cloudtest    16016 2022-10-18 16:10 System.IO.FileSystem.dll
```

### Error details

```console
All files except for pwsh should not have the executable permission set
```

### Environment data

```powershell
Version 7.3.1 for Linux
```

**Comment by rhubarb-geek-nz** [comment]
**Not** fixed in 7.3.2

```
$ curl --silent --location https://github.com/PowerShell/PowerShell/releases/download/v7.3.2/powershell-7.3.2-linux-arm64.tar.gz | tar tvfz - | grep System.dll
-rwxr--r-- cloudtest/cloudtest   49800 2023-01-18 23:01 ref/System.dll
-rwxr--r-- cloudtest/cloudtest   16008 2023-01-18 23:01 ref/System.IO.Compression.FileSystem.dll
-rwxr--r-- cloudtest/cloudtest   16000 2023-01-18 23:01 ref/System.IO.FileSystem.dll
-rwxr--r-- cloudtest/cloudtest    50816 2022-12-06 18:53 System.dll
-rwxr--r-- cloudtest/cloudtest    16016 2022-12-06 18:53 System.IO.Compression.FileSystem.dll
-rwxr--r-- cloudtest/cloudtest    16000 2022-12-06 18:53 System.IO.FileSystem.dll
```

**Comment by rhubarb-geek-nz** [comment]
Not fixed in 7.3.3

```
$ curl --silent --location https://github.com/PowerShell/PowerShell/releases/download/v7.3.3/powershell-7.3.3-linux-arm64.tar.gz | tar tvfz - | grep System.dll
-rwxr--r-- cloudtest/cloudtest   16000 2023-02-18 21:31 ref/System.IO.Compression.FileSystem.dll
-rwxr--r-- cloudtest/cloudtest   16000 2023-02-18 21:31 ref/System.IO.FileSystem.dll
-rwxr--r-- cloudtest/cloudtest   49800 2023-02-18 21:31 ref/System.dll
```
